### ISSUE TITLE
Cannot delete Rancher catalog apps when their creator user is deleted

### ISSUE THREAD

**Issue by xaralis** [rancher/rancher#16274]
**What kind of request is this (question/bug/enhancement/feature request):**

bug

**Steps to reproduce (least amount of steps as possible):**

Create a catalog app deployment, then delete the original user as someone else. Now, with the other user, try to delete or upgrade the app installation.

**Result:**

App cannot be upgraded nor deleted (clone works OK) with following message:

```
Users.management.cattle.io "user-cpf9t" not found; waiting on helm-controller_local
```

Rancher pod logs show:

```
2018/10/24 13:42:17 [ERROR] AppController p-tmfn2/tpsk [helm-controller] failed with : users.management.cattle.io "user-cpf9t" not found
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.1.1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VMs, 4 node setup
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:46:06Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:43:26Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by xaralis** [comment]
For the record, it's possible to work around this issue by manually creating the user using kubectl and setting the fixed ID that is missing:

```
# get some existing user
kubectl get users.management.cattle.io u-6v928 -o yaml > tmpuser.yaml
```

Edit the yaml file, replace the ID with the deleted user's one. Then, create the user again running:

```
kubectl create -f tmpuser.yaml
```

**Comment by eimanavicius** [comment]
Tried your workaround but got error on first step

```
$ kubectl get users.management.cattle.io u-123abc -o yaml > tmpuser.yaml
error: the server doesn't have a resource type "users"
```

**Comment by Odilhao** [comment]
Tried the workaround with rancher v2.1.1 and got the following error message:

```
$ kubectl get users.management.cattle.io u-some-user -o yaml > tmpuser.yaml
error: the server doesn't have a resource type "users"
``` 
There's another way to fix this now?

**Comment by loganhz** [comment]
Delete app issue will be fixed in 2.2.
Created a new ticket for the upgrade issue https://github.com/rancher/rancher/issues/17265

**Comment by jiaqiluo** [comment]
The bug is reproduced on Rancher:v2.1.1 as steps in the original ticket.

Bug fix is validated on Rancher: master 1/3 as steps in the original ticket.
Result: the catalog is removed successfully.
